src/common/event-recorder/src/recorder.rs

```diff
@@ -108,80 +111,75 @@ pub trait Eventable: Send + Sync + Debug {
     }
 }
 
-/// Returns the hints for the insert operation.
-pub fn insert_hints() -> Vec<(&'static str, &'static str)> {
-    vec![
-        (
-            TTL_KEY,
-            EVENTS_TABLE_TTL
-                .get()
-                .map(|s| s.as_str())
-                .unwrap_or(DEFAULT_EVENTS_TABLE_TTL),
-        ),
-        (APPEND_MODE_KEY, "true"),
-    ]
-}
-
-/// Builds the row inserts request for the events that will be persisted to the events table.
-pub fn build_row_inserts_request(events: &[Box<dyn Event>]) -> Result<RowInsertRequests> {
-    // Aggregate the events by the event type.
+/// Aggregates events by its `event_type`.
+#[allow(clippy::borrowed_box)]
+pub fn aggregate_events_by_type(events: &[Box<dyn Event>]) -> HashMap<&str, Vec<&Box<dyn Event>>> {
     let mut event_groups: HashMap<&str, Vec<&Box<dyn Event>>> = HashMap::new();
```

Comment:
Not blocking: maybe we can use `itertools::Itertools` 

https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.chunk_by